docs/howto/how-to-make-api-tool/api-tool-guide.md

```diff
@@ -0,0 +1,341 @@
+        
+        aggregate_tables = [cat, means, mins, maxs]
+
+        return pa.table(aggregate_tables, names = ["category", "mean", "min", "max", ])
+```
+
+This function calls the compute function `pc.min_max()` and `pc.mean()` on each statistical category of the `pa.Table` that we created in step 2. It then aggregates all of these into a new table and returns that table.
+
+### 4. Putting it all together
+Finally, we combine everything we did in steps 1-3 into the `on_complete` function and write the results to the output anchor. It should look like this:
+
+```python
+    def on_complete(self) -> None:
+        """
+        In this method, a Plugin designer should perform any cleanup for their plugin.
+        However, if the plugin is an input-type tool (it has no incoming connections),
+        processing (record generation) should occur here.
+        """
+        import pyarrow as pa
+        import pyarrow.compute as pc
+
+        LEBRON_JAMES_PLAYER_ID = 237
+        NBA_SEASON = 2016
+        lbj_stats = get_postseason_stats(LEBRON_JAMES_PLAYER_ID, NBA_SEASON)
+        table = get_pyarrow_table_from_stats(lbj_stats)
+        output_table = compute_aggregate_output(table)
+        self.provider.write_to_anchor("Output", output_table)
+        self.provider.io.info("APITool tool done.")
+```
+
+And now you're done with writing the code!
+## Package into a YXI 
+---
+Back in our plugin workspace, run the `ayx_plugin_cli create-yxi` command which bundles all the plugins in the workspace into a `.yxi` archive. It should look something like this:
+```powershell
+$ ayx_plugin_cli create-yxi
+[Creating YXI] started
+[Creating YXI] -- generate_config_files:generate_config_xml
+[Creating YXI] -- generate_config_files:generate_tool_config_xml
+[Creating YXI] .  generate_config_files:generate_manifest_jsons
+[Creating YXI] Generating manifest.json for APItool...
+[Creating YXI] Done!
+[Creating YXI] .  generate_artifact:build_artifacts
+[Creating YXI] Creating APItool.yxi...
+[Creating YXI] Creating shiv artifact...
+[Creating YXI] [Installing local dependencies]: python -m pip install -r requirements-local.txt --upgrade --target ~\sdk-api-tool\.ayx_cli.cache\dist
+[Creating YXI] [Compiling shiv artifact]: shiv --compile-pyc --reproducible --extend-pythonpath --site-packages ~\sdk-api-tool\.ayx_cli.cache\dist -o ~\sdk-api-tool\main.pyz -e ayx_python_sdk.providers.amp_provider.__main__:main
+[Creating YXI] Created shiv artifact at: ~\sdk-api-tool\main.pyz
+[Creating YXI] .  create_yxi:create_yxi
+[Creating YXI] finished
+```
+## Run Against Test Client
+---
+After creating the `.yxi` and before installing into Designer, we can first run our plugin against the Test Client to make sure there are no errors and that our plugin will run as expected. 
+Doing so speeds up development time, since we don't have to package and install the `.yxi` into Designer and create a new workflow everytime we make changes to our plugin code. 
```

Comment:
@ayx-orion Update to:

After we create the YXI and before we install it into Designer, we can first run our plugin against the Test Client to make sure there are no errors and that our plugin runs as expected. 
Doing so speeds up development time since we don't have to package and install the YXI into Designer and create a new workflow every time we make changes to our plugin code.